bin/web.rs

```diff
@@ -0,0 +1,277 @@
+            .expect("should be able to bind keyup listener");
+        closure.forget();
+    }
+}
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console, js_name = log)]
+    pub fn log(a: &str);
+
+    #[wasm_bindgen(js_namespace = Date, js_name = now)]
+    pub fn now() -> f64;
+
+    #[wasm_bindgen(js_namespace = window, js_name = getFile)]
+    pub fn get_js_file(file: &str) -> String;
+}
+
+fn create_file(file: &str, buf: &[u8]) {
+    use std::{fs::File, io::Write};
+
+    info!("Creating file {}", file);
+    File::create(file)
+        .expect(&format!("Unable to create {}", file))
+        .write(buf)
+        .expect(&format!("Unable to write in {}", file));
+}
+
+#[macro_use]
+mod rsrc {
+    macro_rules! get_bundled_file {
+        () => {
+            "../res_linux/"
+        };
+    }
+
+    macro_rules! create_file {
+        ($path:expr, $src_dir:expr) => {{
+            create_file($path, include_bytes!(concat!($src_dir, $path)));
+        }};
+    }
+}
+
+pub fn create_fs() {
+    info!("Creating fs");
+
+    let js_files = [
+        // vange-rs files
+        "config/settings.ron",
+        "res/shader/quat.inc.glsl",
+        "res/shader/debug.wgsl",
+        "res/shader/object.wgsl",
+        "res/shader/encode.inc.glsl",
+        "res/shader/quat.inc.wgsl",
+        "res/shader/globals.inc.wgsl",
+        "res/shader/downsample.glsl",
+        "res/shader/body.inc.wgsl",
+        "res/shader/shape.inc.glsl",
+        "res/shader/surface.inc.wgsl",
+        "res/shader/body.inc.glsl",
+        "res/shader/globals.inc.glsl",
+        "res/shader/debug_shape.glsl",
+        "res/shader/shadow.inc.wgsl",
+        "res/shader/surface.inc.glsl",
+        "res/shader/color.inc.wgsl",
+        "res/shader/terrain/locals.inc.wgsl",
+        "res/shader/terrain/ray.wgsl",
+        "res/shader/terrain/paint.wgsl",
+        "res/shader/terrain/mip.wgsl",
+        "res/shader/terrain/scatter.wgsl",
+        "res/shader/terrain/slice.wgsl",
+        "res/shader/physics/collision.inc.glsl",
+        "res/shader/physics/collision_clear.glsl",
+        "res/shader/physics/body_step.glsl",
+        "res/shader/physics/collision_add.glsl",
+        "res/shader/physics/body_push.glsl",
+        "res/shader/physics/pulse.inc.glsl",
+        "res/shader/physics/body_gather.glsl",
+    ];
+
+    for next in js_files {
+        create_file(next, get_js_file(next).as_bytes());
+    }
+
+    #[cfg(not(feature = "nodata"))] {
+        // vangers resouces
+        create_file!("data/escaves.prm", get_bundled_file!());
+        create_file!("data/common.prm", get_bundled_file!());
+        create_file!("data/bunches.prm", get_bundled_file!());
+        create_file!("data/game.lst", get_bundled_file!());
+        create_file!("data/car.prm", get_bundled_file!());
+        create_file!("data/wrlds.dat", get_bundled_file!());
+
+        // the chain
+        create_file!("data/thechain/threall/world.ini", get_bundled_file!());
```

Comment:
We should probably do it in `build.rs`, for multiple reasons. First, we could drive this by data or environment instead of a cargo feature, which means there is less enabling/disabling of code paths going on.
And secondly, most importantly, we could just walk the directory tree and pick up all the files we need, instead of enumerating every single filename/path.